src/test/burn/WixTestTools/RuntimePrereqFeatureFactAttribute.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the Microsoft Reciprocal License. See LICENSE.TXT file in the project root for full license information.
+
+namespace WixTestTools
+{
+    using System;
+    using System.Linq;
+    using System.Management;
+    using System.Collections.Generic;
+    using System.Security.Principal;
+    using WixInternal.TestSupport.XunitExtensions;
+
+    public class RuntimePrereqFeatureFactAttribute : RuntimeFactAttribute
+    {
+        public static HashSet<string> OptionalFeatures = new();
+        public static HashSet<string> ServerFeatures = new();
+        static RuntimePrereqFeatureFactAttribute()
+        {
+            try
+            {
+                var objMC_SF = new ManagementClass("Win32_ServerFeature");
+                var objMOC_SF = objMC_SF?.GetInstances();
+                if (objMOC_SF is not null)
+                {
+                    foreach (var objMO in objMOC_SF)
+                    {
+                        string featureName = (string)objMO.Properties["Name"].Value;
+                        if ((uint)objMO.Properties["InstallState"].Value == 1)
+                        {
+                            ServerFeatures.Add(featureName);
+                        }
+                    }
+                }
+            }
+            catch
+            {
+            }
+
+            try
```

Comment:
This block looks to be a duplicate of the one above. Might be good to create a private static function to take in the management class name and hash set to populate. There is enough code here to reduce the duplication.

Reply:
done